File: Php/core/Validation/SingleValidator.php
<?php
namespace core\Validation{

	abstract class SingleValidator extends BaseValidator{

		public function isValid(){
			$this->ensureErrorArray();
			$this->validate();
			return count($this->errors) == 0;
		}

		protected function isNullOrEmpty($str){
			return $str == null || is_string($str) && strlen($str) == 0;
		}

		protected function isEmail($email){
			return preg_match('/^[a-z0-9&\'\.\-_\+]+@[a-z0-9\-]+\.([a-z0-9\-]+\.)*+[a-z]{2}/is', $email) > 0;
		}

	}

}

File: Web/Validators/SingleValidator.cs
namespace Web.Validators{

	abstract class SingleValidator : BaseValidator{

		public override bool IsValid(){
			EnsureErrorArray();
			Validate();
			return Errors.Count == 0;
		}

		protected bool IsNullOrEmpty(string str){
			return string.IsNullOrEmpty(str);
		}

		protected bool IsEmail(string email)
		{
            //todo: fix the regex
            /*
		    var regex = new Regex("/^[a-z0-9&\'\.\-_\+]+@[a-z0-9\-]+\.([a-z0-9\-]+\.)*+[a-z]{2}/is");
		    return regex.Match(email).Success;
            */
		    return true;
		}

	}

}